pkg/apis/v1/nodepool.go

```diff
@@ -129,6 +130,7 @@ const (
 )
 
 // DisruptionReason defines valid reasons for disruption budgets.
+// CloudProviders will need to append to the list of enums when implementing cloud provider disruption reasons
 // +kubebuilder:validation:Enum={Underutilized,Empty,Drifted}
```

Comment:
I'm not sure what the best option would be here for disruption reason validation. In my mind I see we have two options: 
1. Leave the OpenAPI validation and require cloudprovider to expand the disruption reason as part of the CRD validation
2. Remove the OpenAPI validation and use Runtime validation for the nodepool

I'm leaning towards option 1, since we have precedent of requiring cloud-providers to add additional validation for labels.